browser/ui/ai_chat/ai_chat_tab_helper_unittest.cc

```diff
@@ -0,0 +1,220 @@
+#include "components/favicon/core/test/mock_favicon_service.h"
+#include "content/public/test/test_renderer_host.h"
+#include "content/public/test/web_contents_tester.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using ::testing::NiceMock;
+
+namespace ai_chat {
+
+class MockPrintPreviewExtractor
+    : public AIChatTabHelper::PrintPreviewExtractionDelegate {
+ public:
+  MockPrintPreviewExtractor() = default;
+  ~MockPrintPreviewExtractor() override = default;
+
+  MOCK_METHOD(void, Extract, (bool, ExtractCallback), (override));
+};
+
+class MockPageContentFetcher
+    : public AIChatTabHelper::PageContentFetcherDelegate {
+ public:
+  MockPageContentFetcher() = default;
+  ~MockPageContentFetcher() override = default;
+
+  MOCK_METHOD(void,
+              FetchPageContent,
+              (std::string_view, FetchPageContentCallback),
+              (override));
+  MOCK_METHOD(void,
+              GetSearchSummarizerKey,
+              (mojom::PageContentExtractor::GetSearchSummarizerKeyCallback),
+              (override));
+};
+
+class AIChatTabHelperUnitTest : public content::RenderViewHostTestHarness {
+ public:
+  AIChatTabHelperUnitTest() = default;
+  ~AIChatTabHelperUnitTest() override = default;
+
+  void SetUp() override {
+    content::RenderViewHostTestHarness::SetUp();
+    favicon::ContentFaviconDriver::CreateForWebContents(web_contents(),
+                                                        &favicon_service_);
+    AIChatTabHelper::CreateForWebContents(
+        web_contents(), is_print_preview_supported_
+                            ? std::make_unique<MockPrintPreviewExtractor>()
+                            : nullptr);
+    helper_ = AIChatTabHelper::FromWebContents(web_contents());
+    helper_->SetPageContentFetcherDelegateForTesting(
+        std::make_unique<MockPageContentFetcher>());
+    page_content_fetcher_ = static_cast<MockPageContentFetcher*>(
+        helper_->GetPageContentFetcherDelegateForTesting());
+    print_preview_extractor_ = static_cast<MockPrintPreviewExtractor*>(
+        helper_->GetPrintPreviewExtractionDelegateForTesting());
+  }
+
+  std::unique_ptr<content::BrowserContext> CreateBrowserContext() override {
+    return std::make_unique<TestingProfile>();
+  }
+
+  void NavigateTo(GURL url) {
+    NavigateAndCommit(url);
+    EXPECT_EQ(helper_->GetPageURL(), url);
+  }
+
+  void GetPageContent(ConversationHandler::GetPageContentCallback callback,
+                      std::string_view invalidation_token) {
+    helper_->GetPageContent(std::move(callback), invalidation_token);
+  }
+
+ protected:
+  NiceMock<favicon::MockFaviconService> favicon_service_;
+  raw_ptr<AIChatTabHelper> helper_;
+  raw_ptr<MockPrintPreviewExtractor> print_preview_extractor_;
+  raw_ptr<MockPageContentFetcher> page_content_fetcher_;
+  bool is_print_preview_supported_ = true;
+};
+
+class AIChatTabHelperUnitTest_NoPrintPreview : public AIChatTabHelperUnitTest {
```

Comment:
On master you didn't use a test suite, we have a single test with all the variations https://github.com/brave/brave-core/blob/60cad78225b8727ae291ca1f7cbfde417557c2d7/components/ai_chat/core/browser/conversation_driver_unittest.cc#L448. I created this extra test class just so we could add some tests to verify some functionality with print preview disabled.

And I think it's debatable since not all print preview tests apply to print preview being disabled, and having conditionals on expected functions and expected results adds complexity inside test bodies that can lead to test bugs. 

But I've been thinking about it and I do understand the preference to see the variation in a single test, and see value running all tests with all combinations of params. So I've pushed that change. b234336d95a7fad94b3e55af3af0c087f74e9a8f